### ISSUE TITLE
Phones tags stopped working with TB 102

### ISSUE THREAD

**Issue by cianoz** [thundernest/quicktext#309]
Hello,
just installed Thunderbird 102 and found that now the phones tags I've always used are not recognized anymore.
I refer to [[FROM=phone]] and [[FROM=cellularnumber]]
These 2 tags retrieved the phone numbers from the addressbook, but now with TB 102 they stopped working.
Maybe the fields names are changed, but I couldn't find any reference for this in Mozilla documentation.
I'm using version 5.0 of Quicktext.

**Comment by jobisoft** [comment]
The fields are no longer exposed, it is all stored in the vCard now. Hm.

**Comment by jobisoft** [comment]
[quicktext_5_1.zip](https://github.com/jobisoft/quicktext/files/9033678/quicktext_5_1.zip)

Is this fixing it for you?

**Comment by jobisoft** [comment]
Nope, this one:
[quicktext511.zip](https://github.com/jobisoft/quicktext/files/9033685/quicktext511.zip)

**Comment by cianoz** [comment]
Hello John, thanks for the quick reply
Unfortunately the 511 version doesn't work here. After installing it the Quicktext bar on the new message window disappeared. 
My templates are still there but I can't call then in any way (no bar, no right click item

![immagine](https://user-images.githubusercontent.com/13659070/177031912-2a25d109-7e42-4ddc-9db1-e09c8d7a4a16.png)

**Comment by jobisoft** [comment]
Sorry. Please try 5.1.2:
[quicktext_512.zip](https://github.com/jobisoft/quicktext/files/9034375/quicktext_512.zip)

**Comment by cianoz** [comment]
With the 5.1.2 the bar has come back, but phones numbers are still blank
Since it's Sunday I'd say you'd have to enjoy your weekend John :-)

**Comment by jobisoft** [comment]
Can you try one of these property names:
https://searchfox.org/comm-central/source/mailnews/addrbook/modules/VCardUtils.jsm#308-350

**Comment by cianoz** [comment]
Tried, but with no result
Here is my test template:
`[[FROM=firstname]] [[FROM=lastname]]<br>
[[FROM=custom1]]
tel. [[FROM=WorkPhone]] <br>
mob. [[FROM=CellularNumber]]`

_firstname_ and _lastname_ work, the other tags don't

**Comment by jobisoft** [comment]
Works here:
![image](https://user-images.githubusercontent.com/5830621/177035636-1d0aa482-b7bb-4371-baca-b589d9cea404.png)

Hm. Do you have cardbook installed?

**Comment by cianoz** [comment]
Yes, I'm using CardBook along with the TB integrated addressbook
I just tried on another PC without CardBook and here QuickText  works.
Sorry for the complication
Up until now (with TB 91) I used CardBook along with the native addressbook and it worked. Maybe a bit of tuning and test are needed now with TB 102 to use both together.
Thanks for your readiness and your great work.

**Comment by cianoz** [comment]
It looks that another detail has changed, to take note of: you have to use proper case writing field names / tags
Now "_WorkPhone_" is recognized while "_workphone_" is not

**Comment by jobisoft** [comment]
Cardbook is triggering a different code path and tries to request these values through a cardbook API. I have to look at that.

I am currently adding the same code updates to the TO field collector.

Since it is a limited number of props, I could fix the casing manually (all those linked in https://github.com/jobisoft/quicktext/issues/309#issuecomment-1173041561)

I will report back.

**Comment by jobisoft** [comment]
[quicktext_5.1.3.zip](https://github.com/jobisoft/quicktext/files/9037643/quicktext_5.1.3.zip)

This should now work with the old casing again:

```
[[FROM=firstname]] [[FROM=lastname]]<br> [[FROM=custom1]] tel. [[FROM=workphone]] <br> mob. [[FROM=cellularnumber]]

[[TO=firstname]] [[TO=lastname]]<br> [[TO=custom1]] tel. [[TO=workphone]] <br> mob. [[TO=cellularnumber]]
```
Can you confirm?

Looking at cardbook next.

**Comment by jobisoft** [comment]
CardBook: If the cardbook contact has the requested fields, this version should now work. Please also try after restarting Thunderbird:

[quicktext_5.1.4.zip](https://github.com/jobisoft/quicktext/files/9037844/quicktext_5.1.4.zip)

Does that work for you, @cianoz ?

I think cardbook fails to properly convert TB contacts to CardBook contacts during the initial import, but this should not be related to what @cianoz observed.

**Comment by jobisoft** [comment]
released as 5.2 *fingers crossed*

**Comment by SamuelPlentz** [comment]
IMO close.
